### ISSUE TITLE
'timeout_decorator' library doesn't support Windows

### ISSUE THREAD

**Issue by jmwhitworth** [JMWhitworth/zomboid_rcon#9]
This is mentioned in the README.md, but I'm listing it here additionally in case anyone has any input on potential alternatives.

The timeout decorator in use causes an error that stops this from being used on Windows. This is due to the underlying timing functions. An alternative version is this fork which might work but I don't have a Windows computer to test this on: https://github.com/bitranox/wrapt_timeout_decorator

Please feel free to open a pull request if you have a Windows computer and want to tackle this.

**Comment by xLasttDev** [comment]
Yes i have a windows OS, how to implement this with your module?

**Comment by jmwhitworth** [comment]
You could run it through Docker or WSL, both of which will allow you to run your code through the a Linux/Unix environment. Alternatively, the `timeout_decorator` library could be written out so it's not used, but this would hurt the functionality as it'd no longer handle timeouts nicely.

**Comment by onyx-and-iris** [comment]
Hi, forgive me if I'm just overlooking something, but this package utilises [rcon](https://github.com/conqp/rcon) right? According to the source/client.py the [Client class](https://github.com/conqp/rcon/blob/e10c6f8d621f918573dc81788282e20c1616a910/rcon/source/client.py#L12) subclasses BaseClient which implements a [timeout property](https://github.com/conqp/rcon/blob/e10c6f8d621f918573dc81788282e20c1616a910/rcon/client.py#L41). Could you not set a timeout on the connection directly?

Alternatively, it seems *args, and **kwargs are [passed directly to the BaseClient](https://github.com/conqp/rcon/blob/e10c6f8d621f918573dc81788282e20c1616a910/rcon/source/client.py#L15) so you could pass timeout kwarg to the [Client initialiser](https://github.com/jmwhitworth/zomboid_rcon/blob/a17647746105d942e04ad9203e53c91f04ff22c7/zomboid_rcon/source/RconClient.py#L36).

```python
Client(self._ip, self._port, passwd=self._password, timeout=10)
```

**Comment by jmwhitworth** [comment]
Hi @onyx-and-iris, thanks for looking into this.

I wasn't aware of the built-in timeout functionality when I'd put this package together, so thanks for letting me know. I can try to pick this up in the near future to implement this change, but I'm not certain as to when that be.

In the meantime, if yourself/anyone else who encounters this open issue wishes to put in a merge request to rectify the issue, I'd be most grateful. But there's no expectations at all, of course.

Thanks again.

**Comment by onyx-and-iris** [comment]
Unfortunately I don't even own this game so it makes no sense for me to submit a PR but the change should be minimal:

```python
    def createClient(self) -> Client:
        """Returns an rcon.source.Client object for requests"""
        return Client(self._ip, self._port, passwd=self._password, timeout=10)

    def command(self, command: str, *args) -> CommandResult:
        """Attempts to execute a given command.
        Upon TimeoutError it will retry according to self._retries
        """
        tries = 0
        while tries < self._retries:
            try:
                result = self._command(command, *args)
                return CommandResult(command=command, successful=True, response=result)
            except TimeoutError:
                if self.logging:
                    print(f"({tries+1}/{self._retries}) Request timed out, retrying...")
                tries += 1
        return CommandResult(
            command=command,
            successful=False,
            response=f"Session timed out (after {self._retries} attempt(s))",
        )

    def _command(self, command: str, *args) -> str:
        """Private method to handle timeouts"""
        try:
            with self.createClient() as client:
                return client.run(command, *args)
        except ConnectionRefusedError:
            return "Connection refused"
```

You should be able to rely on the underlying python [socket library to raise a TimeoutError](https://docs.python.org/3/library/socket.html#exceptions). You'll catch it in the same place in {RconClient}.command() so unless I'm overlooking something that's all the changes required. Then you can remove the timeout decorator and the dependency along with it.

It might even be worth making timeout configurable by accepting a timeout kwarg in [ZomboidRCON](https://github.com/jmwhitworth/zomboid_rcon/blob/a17647746105d942e04ad9203e53c91f04ff22c7/zomboid_rcon/zomboid_rcon.py#L10C7-L10C18).

**Comment by jmwhitworth** [comment]
Sorry for the delay on picking this up- but I finally found the time to revisit this and tidy things up.

I just published a new release version `2.0.0` with some big changes, including your approach for removing the `timeout_decorator` library. While your changes from above didn't require a major version change, some of the other improvements did, hence the jump to v2.

I'll mark this as resolved now, but thanks again for your assistance onyx 🙌
